compiler/rustc_hir_analysis/src/lib.rs

```diff
@@ -238,12 +238,18 @@ pub fn check_crate(tcx: TyCtxt<'_>) {
                 check::maybe_check_static_with_link_section(tcx, item_def_id);
             }
             DefKind::Const if !tcx.generics_of(item_def_id).own_requires_monomorphization() => {
-                // FIXME(generic_const_items): Passing empty instead of identity args is fishy but
-                //                             seems to be fine for now. Revisit this!
-                let instance = ty::Instance::new_raw(item_def_id.into(), ty::GenericArgs::empty());
-                let cid = GlobalId { instance, promoted: None };
-                let typing_env = ty::TypingEnv::fully_monomorphized();
-                tcx.ensure_ok().eval_to_const_value_raw(typing_env.as_query_input(cid));
+                let predicates = tcx.predicates_of(item_def_id);
+                let predicates = predicates.instantiate_identity(tcx).predicates;
+
+                if !traits::impossible_predicates(tcx, predicates) {
```

Comment:
Is it guaranteed that `!impossible` means "possible"? Or at least, does the contract for `impossible_predicates` imply that a predicate considered "possible" will not ICE later when one does normal type system things?

(I feel I asked @BoxyUwU that same question already earlier this year in a different context and the answer was "yes it's fine", but I'm not sure...)

Reply:
> my change should in theory eliminate potential ICEs rather than introducing new ones because we eval less.
Footnotes


Yeah, definitely. I was just wondering if it evals less enough. ;)